lib/spack/spack/version.py

```diff
@@ -276,10 +304,23 @@ def satisfies(self, other):
         gcc@4.7 so that when a user asks to build with gcc@4.7, we can find
         a suitable compiler.
         """
+        # If we have commits, we require previous versions to match
+        if self.is_commit and other.is_commit and self.commits:
+            version = Version(self.commits[self.version]['prev_version'])
+            other = Version(self.commits[other.version]['prev_version'])
+
+        # Our version is a commit, and the other is not
+        elif self.is_commit and self.commits:
+            version = Version(self.commits[self.version]['prev_version'])
+        elif other.is_commit and self.commits:
+            other = Version(self.commits[other.version]['prev_version'])
+        else:
+            version = self.version
```

Comment:
```suggestion
            version = self.version
            other = other.version
```
and `other.version` -> `other` for the rest of the method

Reply:
I'm having trouble finding where this is because the file is changed locally from these lines, I'll push changes soon and I'll ask if you can point it out again.

Reply:
okay I think this is outdated - I am replacing this with the below!